base/cvd/cuttlefish/host/commands/cvd/selector/instance_database.cpp

```diff
@@ -274,6 +272,31 @@ std::vector<LocalInstance> InstanceDatabase::FindInstances(
   return ret;
 }
 
+Result<std::pair<cvd::Instance, LocalInstanceGroup>>
+InstanceDatabase::FindInstanceWithGroup(const Queries& queries) const {
+  auto param = CF_EXPECT(FindParam::FromQueries(queries));
+  return viewer_.WithSharedLock<std::pair<cvd::Instance, LocalInstanceGroup>>(
+      [&param](const auto& data)
+          -> Result<std::pair<cvd::Instance, LocalInstanceGroup>> {
+        std::optional<std::pair<cvd::Instance, LocalInstanceGroup>> result_opt;
+        for (const auto& group : data.instance_groups()) {
+          if (!param.Matches(group)) {
+            continue;
+          }
+          for (const auto& instance : group.instances()) {
+            if (!param.Matches(instance)) {
+              continue;
+            }
+            CF_EXPECT(!result_opt.has_value(), "Found more than one instance");
+            result_opt = {instance,
+                          CF_EXPECT(LocalInstanceGroup::Create(group))};
+          }
+        }
+        CF_EXPECT(result_opt.has_value(), "Found no matches");
+        return *result_opt;
```

Comment:
`CF_EXPECT` work with `std::optional`, so this could be

```c++
return CF_EXPECT(result_opt, "Found no matches");
```

Reply:
Done